pkg/controller/kubelet-config/kubelet_config_bootstrap.go

```diff
@@ -0,0 +1,111 @@
+package kubeletconfig
+
+import (
+	"encoding/json"
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	"github.com/openshift/machine-config-operator/pkg/version"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// RunKubeletBootstrap generates MachineConfig objects for mcpPools that would have been generated by syncKubeletConfig
+func RunKubeletBootstrap(templateDir string, kubeletConfigs []*mcfgv1.KubeletConfig, controllerConfig *mcfgv1.ControllerConfig, features *configv1.FeatureGate, mcpPools []*mcfgv1.MachineConfigPool) ([]*mcfgv1.MachineConfig, error) {
+	var res []*mcfgv1.MachineConfig
+	managedKeyExist := make(map[string]bool)
+	userDefinedSystemReserved := make(map[string]string)
+	// Validate the KubeletConfig CR if exists
+	for _, kubeletConfig := range kubeletConfigs {
+		if err := validateUserKubeletConfig(kubeletConfig); err != nil {
+			return nil, newForgetError(err)
```

Comment:
Minor: I see this newForgetError used only for this validation and only in kubeletconfigcontroller one other place. Since its just a wrapper in the other place (not sure why it was needed in the first place) and we don't sync status here, we can probably just return err directly